packages/content/src/lib/content.ts

```diff
@@ -20,19 +22,51 @@ export function injectContent<
   fallback = 'No Content Found'
 ): Observable<ContentFile<Attributes | Record<string, never>>> {
   const route = inject(ActivatedRoute);
-  const contentFiles = injectContentFiles<Attributes | Record<string, never>>();
+  const contentFiles = inject(CONTENT_FILES_TOKEN);
   return route.paramMap.pipe(
     map((params) => params.get(param)),
-    map((slug) => {
-      return (
-        contentFiles.find(
-          (file) => file.filename === `/src/content/${slug}.md`
-        ) || {
+    switchMap((slug) => {
+      const filename = `/src/content/${slug}.md`;
+      const contentFile = contentFiles[filename];
+
+      if (!contentFile) {
+        return of({
           attributes: {},
-          filename: '',
+          filename,
           content: fallback,
+        });
+      }
+
+      return new Promise<string>((resolve) => {
+        if (import.meta.env.SSR === true) {
+          const macroTask = (globalThis as any)[
```

Comment:
This code is needed for SSR because the async call to resolve the content won't be picked as a pending task. This forces a macroTask so the app is "unstable" until the content is resolved.